### PR TITLE
Feature/disallow login to pupil spa if localstorage is disabled

### PR DESCRIPTION
**Pull request by jon-shipley**
# Add local storage check

- [x] trap errors raised by our wrapper function in `storageService`
- [x] add local storage check and redirect to a specialised error page in the 'Sign-in' component
- [x] add local storage check to the '/local-storage-error' route so a page refresh will redirect to '/sign-in' if local storage is enabled
- [x] final QA and cross-browser checks

Screenshot 2021-11-01 at 17 07 37

### CODE REVIEW

File: pupil-spa/src/app/app.module.ts [hunk 1]
```diff
@@ -83,6 +83,7 @@ import { WebsiteOfflineComponent } from './website-offline/website-offline.compo
 import { ApplicationInsightsService } from './services/app-insights/app-insights.service';
 import { TestErrorComponent } from './test-error/test-error.component';
 import { HttpService } from './services/http/http.service';
+import { ErrorLocalstorageComponent } from './error-localstorage/error-localstorage.component';
```
Comment (GuyHarwood) [inline pupil-spa/src/app/app.module.ts:86]: uppercase s on storage?

**Review by GuyHarwood — COMMENTED** [review]
LGSF - just one minor comment

File: pupil-spa/src/app/app.module.ts (hunk 1 above)
Comment (jon-shipley) [reply pupil-spa/src/app/app.module.ts]: Updated

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**